lib/streamlit/elements/lib/column_types.py

```diff
@@ -397,15 +411,27 @@ def NumberColumn(
         Specifies the default value in this column when a new row is added by
         the user. This defaults to ``None``.
 
-    format: str or None
-        A printf-style format string controlling how numbers are displayed.
-        This does not impact the return value. The following formatters are
-        valid: ``%d``, ``%e``, ``%f``, ``%g``, ``%i``, ``%u``. You can also add
-        prefixes and suffixes, e.g. ``"$ %.2f"`` to show a dollar prefix. If
-        this is ``None`` (default), the numbers are not formatted.
+    format:  str, "plain", "localized", "percent", "dollar", "euro", "accounting", "compact", "scientific", "engineering", or None
```

Comment:
should `iso8601` be included in the list?

Reply:
`iso8601` is only available for datetime, time, & date column but isn't supported for number & progress column.